src/SchedulerSubsystem/Scheduler.java

```diff
@@ -62,6 +79,49 @@ else if (event instanceof DestinationEvent)
             throw new InvalidTypeException("Event type received cannot be handled by this subsystem.");
     }
 
+    /**
+     * Filters destination floors from destination events.
+     *
+     * @param events Set of destination events.
+     * @return Set of destination floors.
+     */
+    private Set<Integer> filterDestinationFloors(Set<DestinationEvent> events) {
```

Comment:
This get rid of the destination information which we do need to check against, detrimental at the moment I would remove this metod.